### PR TITLE
feat(statemanager): handle nulls, non-constructables

### PR DESCRIPTION
**Pull request by glennkitchellcaci**
Improved logging of error conditions. Created utility function for boolean switch on isPersistable(); Updated logic in getPersistable() to log and failover in the recoverable state and log for non-recoverable state

### CODE REVIEW

File: src/os/state/basestatemanager.js [hunk 1]
```diff
@@ -141,17 +141,77 @@ os.state.BaseStateManager.prototype.registerPersistable = function(key, clazz) {
...
 os.state.BaseStateManager.prototype.getPersistable = function(key) {
-  if (key in this.persistableMap_) {
-    return new this.persistableMap_[key]();
+  var instance = null;
+  if (this.isPersistable(key)) { // check for nulls, etc
+    var Clazz = this.persistableMap_[key];
+    try {
+      instance = new Clazz();
+    } catch (e) {
+      var args = [this.log, `Registered persistance class for "${key}"`];
```
Comment (jsalankey) [inline src/os/state/basestatemanager.js:157]: Nit: persistence is misspelled in every log statement as "persist*a*nce"

File: src/os/state/basestatemanager.js [hunk 2]
```diff
@@ -141,17 +141,77 @@ os.state.BaseStateManager.prototype.registerPersistable = function(key, clazz) {
 
 
 /**
- * Gets a registered {@link os.IPersistable} class.
+ * Gets an instance of a registered {@link os.IPersistable} class.
  *
  * @param {string} key The object key
  * @return {os.IPersistable}
+ * @suppress {checkTypes} To allow the catch() failover to exist
```
Comment (jsalankey) [inline src/os/state/basestatemanager.js:148]: Whatever this is suppressing, we should stop doing it and get rid of the statement. Suppress calls should never be used to try to help find an error as they're more likely to cause even more errors.

File: src/os/state/basestatemanager.js (hunk 1 above)
Comment (glennkitchellcaci) [reply src/os/state/basestatemanager.js:157]: Oh good, I thought I was crazy and didn't even check the spelling of the other message in here that I was using as my archetype. Updated my new ones and the old one too.

File: src/os/state/basestatemanager.js (hunk 2 above)
Comment (glennkitchellcaci) [reply src/os/state/basestatemanager.js]: Removed. Also removed the corresponding branching and logging types in the code.

**Pull request merged**